services/web-server/src/entities/Artifact.js

```diff
@@ -3,7 +3,8 @@ module.exports = class Artifact {
     Object.assign(this, data);
     this.taskId = taskId;
     this.isPublic = /^public\//.test(this.name);
-    this.isLog = /^text\/plain*/.test(this.contentType);
+    // bug 1638047 - limit log viewer to public text/plain *.log files
+    this.isLog = /^text\/plain.*/.test(this.contentType) && /\.log$/.test(this.name) && this.isPublic;
```

Comment:
No need to add the last check `&& this.isPublic` to define what is a log. We do a bunch of checks with `this.isPublic` in https://github.com/taskcluster/taskcluster/blob/master/services/web-server/src/loaders/artifacts.js.

Reply:
Hm. Per [bug 1638047](https://bugzilla.mozilla.org/show_bug.cgi?id=1638047), we currently treat private `text/plain` artifacts as logs, which breaks. Are you saying there are already checks to not do that?

Reply:
Adding `&& /\.log$/` doesn't fix this? My earlier comment was only referring to `&& this.isPublic`.

Reply:
Won't the same problem occur when we upload a private artifact named `*.log` ?

Reply:
https://firefox-ci-tc.services.mozilla.com/tasks/c1cPlHmTSmeKDfm81l30rQ#artifacts has a file named `releng/adhoc/test.log`, and it's marked as a log file. I'm logged in, with scopes to read that private artifact, but clicking on it gives me a 401 and saving it gives me an html page rather than the text logfile.

Reply:
Regarding the private artifact, I think this may be a different issue similar to https://github.com/taskcluster/taskcluster/issues/2427.

Reply:
:+1: Do we want to leave it broken until #2427 is fixed? Or is it better to treat it as a standard text/plain artifact that we can view until #2427 is fixed?

Reply:
> Do we want to leave it broken until #2427 is fixed?

Sounds good to me.

> Or is it better to treat it as a standard text/plain artifact that we can view until #2427 is fixed?

I'm not sure adding `&& this.isPublic` would fix the issue given the issue is that the signed url doesn't look to be properly created. The problem seems to be the generated url rather than the log viewer. I suspect something is wrong when generating the bewit.

Reply:
The bewit works for me for non-log private artifacts. I've gone ahead and removed the `&& this.isPublic` check though, just to get the fix sooner for other private text/plain artifacts.